core/src/bls_sigverify/bls_sigverifier.rs

```diff
@@ -291,6 +295,10 @@ impl BLSSigVerifier {
             }
         }
 
+        // Send to RPC service for last voted tracking
+        self.alpenglow_last_voted
```

Comment:
Is this potentially in the critical path?  This function can be potentially expensive as it is using `retain`.